### PR TITLE
Update C++ Simulation to match Java

### PR DESCRIPTION
**Pull request by r4stered**
This is a PR that updates the C++ version of Photonlib to support the new simulation features.

TODO:
- [x] Example project

### CODE REVIEW

**Comment by srimanachanta** [comment]
This wasn't based correctly so make sure to merge with master as some stuff is missing.

**Comment by r4stered** [comment]
> This wasn't based correctly so make sure to merge with master as some stuff is missing.

On my fork it says its ahead by 11 commits with master branch. What is missing?

**Comment by srimanachanta** [comment]
> > This wasn't based correctly so make sure to merge with master as some stuff is missing.
> 
> 
> 
> On my fork it says its ahead by 11 commits with master branch. What is missing?

Sync your master branch with our master branch. There should be a button

File: photon-targeting/src/main/native/include/photon/targeting/PhotonTrackedTarget.h
Comment (srimanachanta) [inline photon-targeting/src/main/native/include/photon/targeting/PhotonTrackedTarget.h:1]: ditto on file change

Comment (r4stered) [reply photon-targeting/src/main/native/include/photon/targeting/PhotonTrackedTarget.h:1]: Fixed in latest commit

**Review by mcm001 — APPROVED** [review]
Imagine actually reviewing PRs

**Pull request merged**